File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/project_settings/ProjectSettingsSettings.java
package com.github.alexandrecarlton.idea.settings.layout.project_settings;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.github.alexandrecarlton.idea.settings.layout.project_settings.modules.ModuleSettings;
import com.github.alexandrecarlton.idea.settings.layout.project_settings.project.ProjectSettings;

import org.immutables.value.Value;

import java.util.List;
import java.util.Optional;

@Value.Immutable
@JsonDeserialize(as = ImmutableProjectSettingsSettings.class)
public interface ProjectSettingsSettings {
  Optional<ProjectSettings> project();

  List<ModuleSettings> modules();
}

File: src/main/kotlin/com/github/alexandrecarlton/idea/settings/layout/project_settings/ProjectSettingsSettings.kt
package com.github.alexandrecarlton.idea.settings.layout.project_settings

import com.fasterxml.jackson.annotation.JsonProperty
import com.github.alexandrecarlton.idea.settings.layout.project_settings.modules.ModuleSettings
import com.github.alexandrecarlton.idea.settings.layout.project_settings.project.ProjectSettings

data class ProjectSettingsSettings(
    @JsonProperty("Project")
    val project: ProjectSettings? = null,

    @JsonProperty("Modules")
    val modules: List<ModuleSettings>? = null
)
